File: BasicProblems/timeseries_generation.jl
# An AR1 timeseries is defined by "x_{t+1} = \alpha x_t + \epsilon_{t+1}"
# where "x_0 = 0" and "t = 0,...,T". The shocks "\epsilon_t" are i.i.d. 
# standard normal. Using "T = 200" and "\alpha = 0, 0.5, 0.9" use Plots.jl to
# plot a timecourse for each of the parameters. 

using Plots

T = 200
alphas = [0, 0.5, 0.9]
label = ["α = 0" "α = 0.5" "α = 0.9"]
X = []

for alpha in alphas
    x = zeros(T)
    for i in 1:T-1
        x[i+1] = alpha * x[i] + randn()
    end
    push!(X, x)
end

# Plot seems very slow for some reason
plot(1:T, X, label=label, lw=1.5)
savefig("output")

File: BasicProblems/timeseries_generation.py
# Python seems much faster for this task...
import numpy as np
from random import gauss
import matplotlib.pyplot as plt


T = 200
alphas = [0, 0.5, 0.9]


if __name__ == '__main__':
    fig = plt.figure()

    for alpha in alphas: 
        X = np.zeros(T)
        for i in range(T-1):
            X[i+1] = alpha * X[i] + gauss(0, 1)
        plt.plot(X, linewidth=1.2)

    plt.legend(['α = ' + str(alpha) for alpha in alphas])
    plt.savefig('output')
